src/tox/config/__init__.py

```diff
@@ -1883,28 +1912,17 @@ def getargvlist(cls, reader, value, replace=True):
 
     @classmethod
     def processcommand(cls, reader, command, replace=True):
-        posargs = getattr(reader, "posargs", "")
-        if sys.platform.startswith("win"):
-            posargs_string = list2cmdline([x for x in posargs if x])
-        else:
-            posargs_string = " ".join([shlex_quote(x) for x in posargs if x])
-
         # Iterate through each word of the command substituting as
         # appropriate to construct the new command string. This
         # string is then broken up into exec argv components using
         # shlex.
         if replace:
             newcommand = ""
             for word in CommandParser(command).words():
-                if word == "{posargs}" or word == "[]":
-                    newcommand += posargs_string
+                if word == "[]":
+                    newcommand += reader.getposargs()
                     continue
-                elif word.startswith("{posargs:") and word.endswith("}"):
```

Comment:
This case wasn't handled correctly in the new code, if `:` was in the middle of this text range, resulting in regression https://github.com/tox-dev/tox/issues/1785